Support ssh_config ProxyCommand

lewisthompson opened the pull request.

pahaz commented:
Interesting proposal. Could you please, describe the your use case in the README?

lewisthompson commented:
Sure I'd be happy to. I'll sort something out a week from now as I'm
currently away on vacation.

On Sunday, 28 June 2015, Pahaz Blinov <notifications@github.com> wrote:

> Interesting proposal. Could you please, describe the your use case in the
> README?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pahaz/sshtunnel/pull/18#issuecomment-116301169>.
>

pahaz commented:
@lewisthompson, Hi! How your vacation? :) 
Do you plan on writing your use case in the README?

And thanks again for your PR.

lewisthompson commented:
Hey @pahaz I've just updated the PR and included some documentation.  Essentially if there is a ProxyCommand present in the user's ssh_config file all SSH traffic will be routed through it, as per the way ssh(1) will do.  In this PR I've added an option to disable this functionality.  It is also possible to override (or specify) ssh_proxy.  I hope the documentation is self explanatory, if not, I tested with the following:

```python
ssh_proxy = paramiko.ProxyCommand('ping google.com')
tunnel = SSHTunnelForwarder(ssh_address_or_host=(ssh_server, ssh_port),
                            ssh_username=ssh_username,
                            ssh_password=ssh_password,
                            remote_bind_address=(remote_server, remote_port),
                            ssh_proxy_enabled=True,
                            ssh_proxy=ssh_proxy)
```

obviously this will time out reading/writing from stdin/stdout of `ping google.com` isn't going to give help us much.  Common examples include `nc` etc.

pahaz commented on sshtunnel.py:
```diff
@@ -603,7 +610,12 @@ def __init__(
+                                  .format(repr(ssh_proxy.cmd)))
+                self._transport = paramiko.Transport(ssh_proxy)
+            else:
+                self._transport = paramiko.Transport(ssh_host, ssh_port)
```
`self._transport = paramiko.Transport` expect sock pair as first argument.

pahaz commented on sshtunnel.py:
```diff
@@ -577,6 +581,9 @@ def __init__(
                 hostname_info.get('identityfile', [None])[0]
             ssh_port = ssh_port if ssh_port else \
                 hostname_info.get('port')
+            if not ssh_proxy and hostname_info.get('proxycommand'):
```
Please change code style: `ssh_proxy = ssh_proxy if ssh_proxy else hostname_info.get('proxycommand')`

pahaz commented:
Great work! Can you code issuse and rebase from master?

lewisthompson commented:
@pahaz Updated.  Apologies it took a while - was away once again.
